core/src/preferences/PreferencesBackend.cpp

```diff
@@ -183,87 +183,123 @@ namespace ledger {
 
         std::string PreferencesBackend::createNewSalt(const std::shared_ptr<api::RandomNumberGenerator>& rng) {
             auto bytes = rng->getRandomBytes(128);
-            return std::string(std::begin(bytes), std::end(bytes));
+            return std::string(bytes.begin(), bytes.end());
         }
 
         void PreferencesBackend::setEncryption(
             const std::shared_ptr<api::RandomNumberGenerator>& rng,
             const std::string& password
         ) {
-            // disable encryption to check whether we have a salt already persisted
-            unsetEncryption();
-
-            auto emptySalt = std::string("");
-            auto pref = getPreferences("__core");
-            auto salt = pref->getString(ENCRYPTION_SALT_KEY, emptySalt);
-
-            if (salt == emptySalt) {
-                // we don’t have a proper salt; create one and persist it for future use
-                salt = createNewSalt(rng);
-                pref->editor()->putString(ENCRYPTION_SALT_KEY, salt)->commit();
-            }
-
-            // create the AES cipher
-            _cipher = AESCipher(rng, password, salt, PBKDF2_ITERS);
+            // setting encryption is akin to resetting with an old password that is empty
+            resetEncryption(rng, "", password);
         }
 
         void PreferencesBackend::unsetEncryption() {
```

Comment:
Something is not clear for me
1. What would happen, if we have encrypted data and aomebody calls unsetEncryption? 
2. What is the proper way to remove encryption?

Reply:
1. The code is pretty straightforward: it disables the the AES cipher and you read garbage. This is “only” needed for unit tests. If you have a better idea I would love to switch to your idea (is there a way to test a private method without going nuts with `friend` shitty C++ concepts? :D)
2. `resetEncryption(old, "")`.

Reply:
I.e. if 2. is too weird, I can add a helper function that does it.